app/models/printables/group.rb

```diff
@@ -1,37 +1,40 @@
 module Printables::Group
-  def classify_for_printing(assets, printer_config)
-    assets.reduce({}) do |memo, asset|
+  def classify_for_printing(printer_config)
+    template_cache = Hash.new { |store, types| store[types] = LabelTemplate.for_type(*types).external_id }
```

Comment:
I've moved this into a cache outside the loop, as otherwise we end up hitting the database (or at least the query cache) once per asset. This ensures it only happens once per asset_type/barcode_type.

This isn't the most efficient option, as we'll keep potentially pulling back the same templates, but it should cover the majority of our cases.

Reply:
I don't follow what it means ```LabelTemplate.for_type(*types)```, is it splitting the types array in arguments? (asking because ```def self.for_type(class_type, barcodetype = 'ean13')```)

Reply:
It's the splat operator: https://thoughtbot.com/blog/ruby-splat-operator

But yep, it essentially splits out the array into two different arguments.

Reply:
ok